File: logipack-hub/hub-web/src/lib/server/hubApi/dto/analytics.ts
export type TimeseriesPointDto = {
	bucket_start: string;
	value: number;
};

export type AdminOverviewResponseDto = {
	total_shipments: number;
	shipments_vs_last_period: number;
	shipments_timeseries: TimeseriesPointDto[];
	total_clients: number;
	clients_vs_last_period: number;
	clients_timeseries: TimeseriesPointDto[];
	total_offices: number;
	offices_vs_last_period: number;
	offices_timeseries: TimeseriesPointDto[];
	total_employees: number;
	assigned_employees: number;
	unassigned_employees: number;
	employees_timeseries: TimeseriesPointDto[];
};

export type EmployeeOverviewResponseDto = {
	active_shipments: number;
	active_vs_last_period: number;
	active_timeseries: TimeseriesPointDto[];
	pending_shipments: number;
	pending_vs_last_period: number;
	pending_timeseries: TimeseriesPointDto[];
	deliveries_today: number;
	deliveries_vs_last_period: number;
	deliveries_timeseries: TimeseriesPointDto[];
};

File: logipack-hub/hub-api/src/dto/analytics.rs
use serde::Serialize;

#[derive(Debug, Serialize)]
pub struct TimeseriesPointDto {
    pub bucket_start: String,
    pub value: i64,
}

#[derive(Debug, Serialize)]
pub struct AdminOverviewResponse {
    pub total_shipments: i64,
    pub shipments_vs_last_period: i64,
    pub shipments_timeseries: Vec<TimeseriesPointDto>,
    pub total_clients: i64,
    pub clients_vs_last_period: i64,
    pub clients_timeseries: Vec<TimeseriesPointDto>,
    pub total_offices: i64,
    pub offices_vs_last_period: i64,
    pub offices_timeseries: Vec<TimeseriesPointDto>,
    pub total_employees: i64,
    pub assigned_employees: i64,
    pub unassigned_employees: i64,
    pub employees_timeseries: Vec<TimeseriesPointDto>,
}

#[derive(Debug, Serialize)]
pub struct EmployeeOverviewResponse {
    pub active_shipments: i64,
    pub active_vs_last_period: i64,
    pub active_timeseries: Vec<TimeseriesPointDto>,
    pub pending_shipments: i64,
    pub pending_vs_last_period: i64,
    pub pending_timeseries: Vec<TimeseriesPointDto>,
    pub deliveries_today: i64,
    pub deliveries_vs_last_period: i64,
    pub deliveries_timeseries: Vec<TimeseriesPointDto>,
}
